### ISSUE TITLE
rspack cli should not print help information when config file is not found and other errors.

### ISSUE THREAD

**Issue by XGHeaven** [web-infra-dev/rspack#4314]
### System Info

In Github CodeSpaces
```
  System:
    OS: Linux 6.2 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 AMD EPYC 7763 64-Core Processor
    Memory: 13.30 GB / 15.61 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 16.18.0 - /usr/local/share/nvm/versions/node/v16.18.0/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.19.2 - /usr/local/share/nvm/versions/node/v16.18.0/bin/npm
  npmPackages:
    @rspack/cli: 0.3.5 => 0.3.5 
    @rspack/plugin-html: 0.3.5 => 0.3.5 
```

### Details

When using a not existing config file. The error messages included the help message that was like from `--help`.

<img width="849" alt="image" src="https://github.com/web-infra-dev/rspack/assets/9291212/4d648c41-df8f-492a-8968-3426b422f492">

### Reproduce Steps

1. Install Rspack cli
2. Run `npx rspack build -c a.js` to run a not existing config file.

**Comment by hardfist** [comment]
it's indeed a bug, are you willing to fix it for us?

**Comment by OceanPresentChao** [comment]
> it's indeed a bug, are you willing to fix it for us?

Would you mind assigning it to me?

**Comment by hardfist** [comment]
thanks

**Comment by hardfist** [comment]
landed in #4322
